File: OCH_Tao/ABC/B/302.py
import numpy as np
H, W = map(int, input().split())
S = np.array([list(input()) for _ in range(H)])
X = np.argwhere(S == "s")
D = [(-1, -1), (-1, 0), (-1, 1), (0, -1), (0, 1), (1, -1), (1, 0), (1, 1)]
for i in X:
  cnt = ""
  for x, y in D:
    for n in range(5):
      if 0 <= i[0]+x*n < H and 0 <= i[1]+y*n < W:
        cnt += S[i[0]+x*n][i[1]+y*n]
      else:
        cnt = ""
        break
    if cnt == "snuke":
      [print(i[0]+x*n+1, i[1]+y*n+1) for n in range(5)]
      exit()
    else:
      cnt = ""

File: OCH_Tao/ABC/B/302.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [H, W] = input.shift().split(" ").map(x => parseInt(x, 10));
const S = input.map(x => [...x]);
const D = [[-1, -1], [-1, 0], [-1, 1], [0, -1], [0, 1], [1, -1], [1, 0], [1, 1]];
for (let i = 0; i < H; i++) {
  for (let j = 0; j < W; j++) {
    if (S[i][j] === "s") {
      let str = "";
      for (const [x, y] of D) {
        for (let n = 0; n < 5; n++) {
          if (0 <= i + x * n && i + x * n < H && 0 <= j + y * n && j + y * n < W) {
            str += S[i + x * n][j + y * n];
          } else {
            break;
          }
        }
        if (str === "snuke") {
          for (let n = 0; n < 5; n++) {
            console.log([i + x * n + 1, j + y * n + 1].join(" "));
          }
          process.exit();
        } else {
          str = "";
        }
      }
    }
  }
}